spec/formatting.md

```diff
@@ -18,3 +18,248 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+Errors in messages and their formatting may occur and be detected
+at multiple different stages of their processing.
+Where available,
+the use of validation tools is recommended,
+as early detection of errors makes their correction easier.
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is not well-formed.
+
+  Example invalid messages resulting in a Syntax error:
+
+  ```
+  {Missing end brace
+  ```
+
+  ```
+  {Unknown {?placeholder}}
+  ```
+
+  ```
+  let $var = {(no message body)}
+  ```
+
+- **Data Model errors** occur when a message is invalid due to
+  violating one of the semantic requirements on its structure:
+
+  - **Variant Key Mismatch errors** occur when the number of keys on a Variant
+    does not equal the number of Selectors.
+
+    Example invalid messages resulting in a Variant Key Mismatch error:
+
+    ```
+    match {$one}
+    when 1 2 {Too many}
+    when * {Otherwise}
+    ```
+
+    ```
+    match {$one} {$two}
+    when 1 2 {Two keys}
+    when * {Missing a key}
+    when * * {Otherwise}
+    ```
+
+  - **Missing Fallback Variant errors** occur when the message
+    does not include a Variant with only catch-all keys.
+
+    Example invalid messages resulting in a Missing Fallback Variant error:
+
+    ```
+    match {$one}
+    when 1 {Value is one}
+    when 2 {Value is two}
+    ```
```

Comment:
Perhaps add text to say:

> Missing _when *_ case:

(and on line 82 add "Missing `when * *` case:")

Reply:
I don't see the necessity for these specifiers. Are these specific examples more difficult to respond about than the others, or should we specify explicitly what's wrong in each of them?

Regarding the examples in general, I at least prefer requiring the reader to at least momentarily pause on some of them to figure out for themselves what's going on, rather than making sure that no thought is required.